### PR TITLE
fix: Updated navbar sections in desktop and mobile view

### PR DESCRIPTION
**Pull request by AbhiDiva96**
## Related issue

fixes: #305 

## Description

In Palettegram, Navbar component has been enhanshed there are several changes taken place such as themeButton sidebar toggler visibility (register or login button)

## Before 
Screenshot 2024-05-15 133926

## After
Screenshot 2024-05-15 133939

## Before
Screenshot 2024-05-15 134003

## After
Screenshot 2024-05-15 134013

## Before
Screenshot 2024-05-15 134026

## After change
Screenshot 2024-05-15 134043

## Summary by CodeRabbit

- **New Features**
  - Added a theme button to toggle between dark and light modes.
  - Introduced a link to the GitHub repository with star count in the navigation bar.

- **Improvements**
  - Enhanced logout functionality to include a redirect after logging out.
  - Updated navigation bar structure for better usability on different screen sizes.
  - Improved mobile menu layout and styling for a better user experience.

### CODE REVIEW

**Comment by yashd-dev** [comment]
A couple of issues that needs to be resolved,  

image
In the above image, the ThemeButton's icon colour has not changed in light mode, thus seeming invisible.

<br>

image
Please Improve the UI of it, and keep it consistent between light and dark mode. The ThemeButton shouldn't really overlap with Github Stars. Here is a reference you can look at and improvise the UI of Navbar https://astroship.web3templates.com/

**Comment by Sanchitbajaj02** [comment]
> ## After
> Screenshot 2024-05-15 133939

I think the previous design was good for desktop and does not require any change

**Comment by Sanchitbajaj02** [comment]
> ## After
> Screenshot 2024-05-15 134013 

A space between icons would be appreciated

**Comment by Sanchitbajaj02** [comment]
image

Why is the sidebar not covering the entire side and just covering the top layer?

**Comment by AbhiDiva96** [comment]
> > ## After
> > Screenshot 2024-05-15 133939
> 
> I think the previous design was good for desktop and does not require any change

ok got it. if it is good then no issue the think had to change is sidebar section.

**Comment by AbhiDiva96** [comment]
> image
> 
> Why is the sidebar not covering the entire side and just covering the top layer?

ok got it. i am resolving this issue

**Comment by Sanchitbajaj02** [comment]
@AbhiDiva96 ALSO UPDATE THIS PR TO THE LATEST COMMIT

File: src/components/core/themeButton/index.tsx [hunk 1]
```diff
@@ -28,20 +28,21 @@ export default function ThemeButton({ iconSize }: { iconSize: number }) {
     <>
       {theme === "dark" ? (
         <button
-          className="px-2 py-2 md:flex fixed md:static block md:top-0 md:m-0 top-1 m-2 rounded-full hover:bg-primary-light hover:scale-105 bg-primary text-white"
+          className="px-2 py-2 md:flex fixed md:static block md:top-0 md:m-0 top-1 m-2 rounded-full hover:bg-primary-light hover:scale-105 border-2 border-white md:bg-transparent md:border-2 md:border-primary text-white"
```
Comment (Sanchitbajaj02) [inline src/components/core/themeButton/index.tsx:31]: roll back to original one, as this component might hamper for different screen sizes. You can please remove the fixed property

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**